scripts/pre_commit_linter.py

```diff
@@ -505,6 +541,27 @@ def _check_newline_character(all_files):
     return summary_messages
 
 
+def _check_bad_pattern_in_file(filename, content, pattern):
+    """
+        Detects whether the given pattern is present in the file.
+        Args:
+            filename: str. Name of the file.
+            content: str. Contents of the file.
+            pattern: dict. Object containing details for the pattern
+                to be checked.
+    """
+    regexp_pattern = pattern['regexp']
+    if not (any(filename.startswith(bad_pattern)
+                for bad_pattern in pattern['excluded_dirs'])
+            or filename in pattern['excluded_files']):
+        if re.search(regexp_pattern, content):
+            failed = True
+            print '%s --> %s' % (
+                filename, pattern['message'])
+            return 1
+    return 0
```

Comment:
Actually it's probably better to check each file and each error and print each occurrence of the error, adding 1 each time. Otherwise the developer will need to run this check, then fix it, then run again to find the next issue, then fix it, etc.

Reply:
I am making change but I'm not sure if I understand fully.

So, please take a look again in next iteration.